### DISCUSSION TITLE
Clarification on 'Use soft keyword in compiler' option needed

### DISCUSSION THREAD

**Discussion by dmitrykolesnikovich** [cashapp/exhaustive#25] [Q&A]
So, is it clear enough how to implement this option for writing it out step by step? Is forking of whole compiler codebase strictly necessary?

**Comment by JakeWharton** [comment]
It's not syntactically valid Kotlin today so yes I believe it requires a fork of the language. We did not explore it beyond discussion so I'm unsure if it could be accomplished with solely a compiler plugin or if an entire compiler for is needed. But I strongly suspect it will require a fork.
